### PR TITLE
OJ-2455: Fix template and refactoring of OTG state machine

### PR DESCRIPTION
**Pull request by surajkumar**
### What and why did it change
* Lambdas were not invoking due a problem with the `template.yaml` not pointing to the Lambda handlers correctly. This has been fixed and the repository has been aligned with ipv-cri-check-hmrc-api. 
* Fixed incorrect substitution in state machine
* Removed the empty pass states because this was causing the state machine to pass on fail. 
* Removed the parallel branch as this is no longer needed as per architectures updates. Now the Bearer Token and Expiry are stored together. 
* Removed all the "non-flow" tests from mocked as they were testing the mocked data rather than the state machine.

Screenshot:
image

### Issue tracking
- [OJ-2455](https://govukverify.atlassian.net/browse/OJ-2455)
- [OJ-2352](https://govukverify.atlassian.net/browse/OJ-2352)

[OJ-2455]: https://govukverify.atlassian.net/browse/OJ-2455?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[OJ-2352]: https://govukverify.atlassian.net/browse/OJ-2352?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: infrastructure/template.yaml [hunk 1]
```diff
@@ -163,21 +163,37 @@ Resources:
...
+      LoggingConfig:
+        LogGroup: !Sub /aws/lambda/${AWS::StackName}/BearerTokenHandlerFunction
       CodeSigningConfigArn:
         !If [EnforceCodeSigning, !Ref CodeSigningConfigArn, !Ref AWS::NoValue]
 
+  BearerTokenHandlerFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/lambda/${AWS::StackName}/BearerTokenHandlerFunction
+      RetentionInDays: !If [IsDevLikeEnvironment, 7, 30]
```
Comment (jkunle) [inline infrastructure/template.yaml:176]: Please can you switch this, like Tim did, to [IF PROD 30, 7]

File: infrastructure/template.yaml [hunk 2]
```diff
@@ -163,21 +163,37 @@ Resources:
...
+      LoggingConfig:
+        LogGroup: !Sub /aws/lambda/${AWS::StackName}/TotpGeneratorFunction
       CodeSigningConfigArn:
         !If [EnforceCodeSigning, !Ref CodeSigningConfigArn, !Ref AWS::NoValue]
 
+  TotpGeneratorFunctionLogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub /aws/lambda/${AWS::StackName}/TotpGeneratorFunction
+      RetentionInDays: !If [IsDevLikeEnvironment, 7, 30]
```
Comment (jkunle) [inline infrastructure/template.yaml:195]: same here

File: infrastructure/template.yaml (hunk 1 above)
Comment (surajkumar) [reply infrastructure/template.yaml:176]: I copied Tim's code from here: https://github.com/govuk-one-login/ipv-cri-check-hmrc-api/blob/5d1924099fc4c604118b47d1ada7ed0214fda436/infrastructure/template.yaml#L206 

Would be better to stay consistent with the other code

File: step-functions/oauth-token-generator.asl.json [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
-        }
-      ]
+      "ResultPath": "$.totpSecret"
     },
-    "Lambda Invoke Generate TOTP Code": {
+    "Fetch HMRC Client Secret": {
+      "Type": "Task",
+      "Next": "Generate TOTP Code",
+      "Parameters": {
+        "SecretId": "${ClientSecretName}"
```
Comment (jkunle) [inline step-functions/oauth-token-generator.asl.json:27]: Please can you use selectors, so we don't get all the other garbage we don't care about filtered out, so we have just to data we need passed down from each state

File: infrastructure/template.yaml (hunk 1 above)
Comment (jkunle) [reply infrastructure/template.yaml:176]: Looks like we are not consistent anyway https://github.com/govuk-one-login/ipv-cri-check-hmrc-api/blob/5d1924099fc4c604118b47d1ada7ed0214fda436/infrastructure/template.yaml#L607
I thought he changed it everywhere, when I saw this I thought this is better do we really need more than 7 days in all the other environments?

File: step-functions/oauth-token-generator.asl.json (hunk 3 above)
Comment (jkunle) [reply step-functions/oauth-token-generator.asl.json:27]: Is there a reason why we can put all the Fetch Secrets in a `parallel block`

**Review by jkunle — APPROVED** [review]

**Pull request merged**